### PR TITLE
[move] IR parse errors must have locations 

### PR DESCRIPTION
**Pull request by modocache**
Use Rust types to disallow surfacing Move IR parse errors that do not have locations. Now, errors such as "duplicate ability" must be displayed with a location (and in this change, that location points to the first duplicate occurrence of an ability).

### CODE REVIEW

**Review by bob-wilson — DISMISSED** [review]
This seems fine.

File: language/compiler/ir-to-bytecode/src/parser.rs [hunk 1]
```diff
@@ -68,24 +68,27 @@ pub fn parse_cmd_(
...
+            Label::primary(id, location.usize_range()).with_message(format!("{}", error))
         }
     };
-    bail!("ParserError: {}", msg)
+    let message = lbl.message.clone();
+    let error = Diagnostic::error()
+        .with_message("Parser Error")
+        .with_labels(vec![lbl]);
+    let writer = &mut StandardStream::stderr(ColorChoice::Auto);
+    emit(writer, &Config::default(), &files, &error).unwrap();
```
Comment (tnowacki) [inline language/compiler/ir-to-bytecode/src/parser.rs:90]: This will likely cause issues, but it was already doing this. 
We should have the option to handle errors and emit to a buffer, makes test runners a lot easier to deal with. You don't want things randomly printing (Again was already doing this so we can do it in another PR)

**Review by vgao1996 — DISMISSED** [review]
Seems fine based on the comments. Restamping
